app/role_data.go

```diff
@@ -34,20 +34,21 @@ func addRoles(management *config.ManagementContext) (string, error) {
 
 	rb.addRole("User", "user").addRule().apiGroups("management.cattle.io").resources("principals", "roletemplates").verbs("get", "list", "watch").
 		addRule().apiGroups("management.cattle.io").resources("users").verbs("get", "list", "watch").
-		addRule().apiGroups("management.cattle.io").resources("preferences").verbs("*").
 		addRule().apiGroups("management.cattle.io").resources("settings").verbs("get", "list", "watch").
 		addRule().apiGroups("management.cattle.io").resources("clusters").verbs("create").
 		addRule().apiGroups("management.cattle.io").resources("templates", "templateversions").verbs("get", "list", "watch").
 		addRule().apiGroups("management.cattle.io").resources("nodedrivers").verbs("get", "list", "watch").
-		addRule().apiGroups("management.cattle.io").resources("podsecuritypolicytemplates").verbs("get", "list", "watch").
-		addRule().apiGroups("management.cattle.io").resources("nodetemplates").verbs("*").
-		addRule().apiGroups("management.cattle.io").resources("sourcecodecredentials").verbs("*").
-		addRule().apiGroups("management.cattle.io").resources("sourcecoderepositories").verbs("*")
+		addRule().apiGroups("management.cattle.io").resources("podsecuritypolicytemplates").verbs("get", "list", "watch")
 
 	rb.addRole("User Base", "user-base").addRule().apiGroups("management.cattle.io").resources("principals", "roletemplates").verbs("get", "list", "watch").
 		addRule().apiGroups("management.cattle.io").resources("users").verbs("get", "list", "watch").
 		addRule().apiGroups("management.cattle.io").resources("preferences").verbs("*").
 		addRule().apiGroups("management.cattle.io").resources("settings").verbs("get", "list", "watch")
+	rb.addRole("User Scope", "user-scoped").
```

Comment:
Where and how will this role be used?

Reply:
Ok, i see now that you indirectly reference it via 
```
+				Name:     clusterrolePrefix + "user-scoped",
```

I would like to do it a different way. First, you do not need to define a role here for it all all. These are assignable globalRoles. We do not need or want this to be an assignable global role.

Just ensure a clusterRole exists at the time of doing the binding. Its ok to do it just based on a hard-coded name like `cattle-user-scoped-resources`.
Check that that roles exist when creating the binding and if it doesnt, create it. Use a cache (LIster) for checking if it exists so that the check is inexpensive.

Reply:
So, just like how rolebinding and namespace have been done.